src/main/java/igrad/model/requirement/Credits.java

```diff
@@ -1,61 +1,109 @@
 package igrad.model.requirement;
 
 import static igrad.commons.util.AppUtil.checkArgument;
+import static igrad.commons.util.CollectionUtil.requireAllNonNull;
 import static java.util.Objects.requireNonNull;
 
 /**
- * Represents a Requirement's required credit units in the course book.
- * Guarantees: immutable, non-null and is valid as declared by {@link #isValidCredits(String)}
+ * Represents a Requirement's credits information in the course book.
+ * Guarantees: immutable, fields are non-null and valid as declared by {@link #isValidCredits(String)}
  */
 public class Credits {
 
     public static final String MESSAGE_CONSTRAINTS =
-        "Credits needed to satisfy requirement should be more than 0.";
+        "Modular credits needed to satisfy requirement should be more than 0.";
 
+    // TODO
     public static final String VALIDATION_REGEX = "^[0-9]\\d*$"; // allows any numbers more than or equals zero
 
-    public final String value;
+    public final String creditsRequired;
```

Comment:
I think its better to represent credits as `Integer` instead of `String` (justification: less parsing needed when doing credits checking creditsRequired - creditsFulfilled)

Reply:
lemme think about it

Reply:
ok hmm, on the contrary, changing it to an internal Integer representation might result in an overall inconsistency (cos all Model classes use String as their `value`)

Reply:
but so far in the codebase, for the Model classes (even in the original i think), we don't have a precedence yet where the entity in question can be entirely represented using Integer (i.e, this is the first such instance), thus it is justifiable to 'violate' the inconsistency in favour of this special case

Reply:
these are just my thoughts when i was looking at this code, you're free to make your own choices!